src/Components/Server/src/DependencyInjection/ComponentServiceCollectionExtensions.cs

```diff
@@ -60,6 +61,12 @@ public static IServerSideBlazorBuilder AddServerSideBlazor(this IServiceCollecti
         // Here we add a bunch of services that don't vary in any way based on the
         // user's configuration. So even if the user has multiple independent server-side
         // Components entrypoints, this lot is the same and repeated registrations are a no-op.
+
+        services.TryAddSingleton(s =>
+        {
+            var meterFactory = s.GetService<IMeterFactory>();
+            return meterFactory != null ? new CircuitMetrics(meterFactory) : null;
+        });
```

Comment:
We should call `services.AddMetrics` the same way we call `services.AddDataProtection()` and just call `services.TryAddSingleton<CircuitMetrics>()` here.

Everywhere else we don't need to conditionally resolve the service or check for null as we can depend on it always being available and initialized.